packages/askar/src/AskarApi.ts

```diff
@@ -0,0 +1,87 @@
+import { AgentContext } from '@credo-ts/core'
+import { injectable } from 'tsyringe'
+
+import { AskarStoreRotateKeyOptions, AskarStoreExportOptions, AskarStoreImportOptions } from './AskarApiOptions'
+import { AskarStoreManager } from './AskarStoreManager'
+
+@injectable()
+export class AskarApi {
+  public constructor(private agentContext: AgentContext, private askarStoreManager: AskarStoreManager) {}
+
+  public get isStoreOpen() {
+    return this.askarStoreManager.isStoreOpen(this.agentContext)
+  }
+
+  /**
+   * @throws {AskarStoreDuplicateError} if the wallet already exists
+   * @throws {AskarStoreError} if another error occurs
+   */
+  public async provisionStore(): Promise<void> {
+    await this.askarStoreManager.provisionStore(this.agentContext)
+  }
+
+  /**
+   * @throws {AskarStoreNotFoundError} if the wallet does not exist
+   * @throws {AskarStoreError} if another error occurs
+   */
+  public async openStore(): Promise<void> {
+    await this.askarStoreManager.openStore(this.agentContext)
+  }
+
+  /**
+   * Rotate the key of the current askar store.
+   *
+   * NOTE: multiple agent contexts (tenants) can use the same store. This method rotates the key for the whole store,
+   * it is advised to only run this method on the root tenant agent when using profile per wallet database strategy.
+   * After running this method you should change the store configuration in the Askar module.
+   *
+   * @throws {AskarStoreNotFoundError} if the wallet does not exist
+   * @throws {AskarStoreError} if another error occurs
+   */
+  public async rotateStoreKey(options: AskarStoreRotateKeyOptions): Promise<void> {
```

Comment:
Askar specific operations will now be part of the askar api. So there's no global rotateWalletKey anymore, as that's specific to Askars way of doing this.